#15 Fix TypeError (String can't be coerced into Integer) on collect_output

JunOkumura opened the pull request:
cast amount from String to Integer.
The amount is like "50.000" that is String and TPC digit.

resolve #15

azuchi commented on lib/glueby/contract/tx_builder.rb:
```diff
@@ -8,7 +8,7 @@ module TxBuilder
       def collect_outputs(results, amount)
         results.inject([0, []]) do |sum, output|
-          new_sum = sum[0] + (output['amount'] * 100_000_000)
+          new_sum = sum[0] + (output['amount'].to_i * 100_000_000)
```
The values returned by listunspent include values after the decimal point, such as `"0.41547730"`. 
In that case, this code output `0` value.

And please implement unit test.

JunOkumura replied:
OK, Thank you review.
I will fix it, and run unit tests.

JunOkumura commented:
I fixed it is reviewed code by you.

I modified code in `lib/glueby/internal/wallet/tapyrus_core_wallet_adapter.rb` for using `tpc_to_tapyrus`.
`tpc_to_tapyrus` move to module that is `Glueby::Internal::Wallet::TapyrusCoreWalletAdapter::Util`, and it change method from private to public.

I also fixed some test datas.
I returned the amount from rpc is String, but test data was Float.
I replaced them to String.

azuchi commented on lib/glueby/contract/tx_builder.rb:
```diff
@@ -1,14 +1,16 @@
       def collect_outputs(results, amount)
         results.inject([0, []]) do |sum, output|
-          new_sum = sum[0] + (output['amount'] * 100_000_000)
+          new_sum = sum[0] + tpc_to_tapyrus(output['amount'])
```
The conversion process itself should be done in `Timestamp::Util#listunspent`.

JunOkumura replied:
ok

JunOkumura commented:
`tpc` and `tapyrus` translation moved to `Timestamp#list_unspent`.
So I do not need to change some mock amounts, revert it. But the amounts had to be translated to `tapyrus`, I replaced from `tpc` to `tapyrus`.

azuchi commented on spec/glueby/contract/timestamp_spec.rb:
```diff
@@ -13,11 +13,11 @@
         'txid' => '0555c5af698db73ed6378d2ed3c71e45fc6a1dbbcf931a248a7c9221f1d3220c',
         'vout' => 0,
-        'amount' => 1.00000000
+        'amount' => 100000000
```
This data should be a string as it comes from the Bitcoin Core RPC `listunspent`.

JunOkumura replied:
I got it.
I will fix soon.

JunOkumura commented:
I changed the amount in `timestamp_spec.rb` to String that is return of `Tapyrus Core RPC`.
